### ISSUE TITLE
tensorrt IPluginV2 can't deserializeCudaEngine

### ISSUE THREAD

**Issue by satchelwu** [NVIDIA/TensorRT#92]
I followed the develop guide to implement the custom layer("PReLU") with IPluginV2, IPluginCreator and IPluginFactoryV2(REGISTER_TENSORRT_PLUGIN(PReLUPluginCreator)). 
There is no problem to run the code bellow and successfully build CudaEngine and serialize to the IHostMemory. 

        NPluginFactory pluginFactory;
	parser->setPluginFactoryV2(&pluginFactory);
	const nvcaffeparser1::IBlobNameToTensor* blobNameToTensor =
		parser->parse("models/det1.prototxt", "models/det1.caffemodel", *network, nvinfer1::DataType::kFLOAT);

but when I want to deserialize the IHostMemory to create cuda engine

	nvinfer1::ICudaEngine* mEngine = runtime->deserializeCudaEngine(pHostMemory, pHostMemory->size(), nullptr);

 there is an error happend, named "can not parse layer type PReLU". I skipped the source code, in the H file "opParsers.h"  

static std::unordered_map<std::string, LayerParseFn> gParseTable 
{
    {"Convolution", parseConvolution},
    {"Pooling", parsePooling},
    {"InnerProduct", parseInnerProduct},
    {"ReLU", parseReLU},
    {"Softmax", parseSoftMax},
    {"SoftmaxWithLoss", parseSoftMax},
    {"LRN", parseLRN},
    {"Power", parsePower},
    {"Eltwise", parseEltwise},
    {"Concat", parseConcat},
    {"Deconvolution", parseDeconvolution},
    {"Sigmoid", parseSigmoid},
    {"TanH", parseTanH},
    {"BatchNorm", parseBatchNormalization},
    {"Scale", parseScale},
    {"Crop", parseCrop},
    {"Reduction", parseReduction},
    {"Reshape", parseReshape},
    {"Permute", parsePermute},
    {"ELU", parseELU},
    {"BNLL", parseBNLL},
    {"Clip", parseClip},
    {"AbsVal", parseAbsVal}
};

one plugin is connected to one parser.  
I can not jump into the IRuntime.deserializeCudaEngine. 
Could anyone help? Thanks

**Comment by satchelwu** [comment]
I reviewed the code， finally the problem was solved.
